src/NServiceBus.Metrics.PerformanceCounters/PerformanceCounterUpdater.cs

```diff
@@ -79,9 +79,17 @@ async Task Cleanup(CancellationToken cancellationToken)
     {
         try
         {
-            while (!cancellationToken.IsCancellationRequested)
+            while (true)
```

Comment:
I think this is better because:

- The next operation after this condition is the `Task.Delay` anyway.
- There is now only one way out of the method. Before, the method _could_ exit silently in the (probably rare) case that the token is triggered after `Task.Delay` returns and before the condition. Now the debug message will _always_ be logged.